xbmc/video/VideoInfoScanner.cpp

```diff
@@ -1792,14 +1793,17 @@ namespace KODI::VIDEO
     {
       if (!pItem->SkipLocalArt())
       {
+        bool useFolder = false;
         if (bApplyToDir && (content == CONTENT_MOVIES || content == CONTENT_MUSICVIDEOS))
         {
-          std::string filename = pItem->GetLocalArtBaseFilename();
+          std::string filename = ART::GetLocalArtBaseFilename(*pItem, useFolder);
           std::string directory = URIUtils::GetDirectory(filename);
           if (filename != directory)
             AddLocalItemArtwork(art, artTypes, directory, addAll, exactName);
         }
-        AddLocalItemArtwork(art, artTypes, pItem->GetLocalArtBaseFilename(), addAll, exactName);
+        useFolder = false;
```

Comment:
Why is this line added? Maybe a comment is needed?

Reply:
it's to ensure the same behavior as earlier. notice that the call to GetLocalArtBaseFilename can mutate the bool parameter. since i'm reusing the same variable i need to reset it false in case it was mutated. earlier there was a convenience wrapper without the bool parameter which just threw it away. as it saw were limited use (basically, just here) i am of the opinion that bloating the api is a bigger problem than the slight convenience offered by the wrapper.

Reply:
Agreed.

How about:

```patch
--- a/xbmc/video/VideoInfoScanner.cpp
+++ b/xbmc/video/VideoInfoScanner.cpp
@@ -1798,7 +1798,11 @@ namespace KODI::VIDEO
           if (filename != directory)
             AddLocalItemArtwork(art, artTypes, directory, addAll, exactName);
         }
+
+        // // Reset useFolder to false as GetLocalArtBaseFilename may modify it
+        // in the previous call.
         useFolder = false;
+
         AddLocalItemArtwork(art, artTypes, ART::GetLocalArtBaseFilename(*pItem, useFolder), addAll,
                             exactName);
       }
```

Reply:
sure.